File: mobile/ZirveMobile/ZirveMobile/Models/Event.swift
import Foundation

struct EventPhoto: Identifiable, Codable {
    let id: UUID
    let event_id: UUID
    let file_path: String
    let original_filename: String
    let caption: String?
    let display_order: Int
    let is_cover: Bool
}

struct Waypoint: Codable, Identifiable {
    let order: Int
    let lat: Double
    let lng: Double
    let label: String

    var id: Int { order }
}

struct Event: Identifiable, Codable, Equatable, Hashable {
    let id: String
    let title: String
    let description: String?
    let category: String
    let difficulty: String
    let location_name: String?
    let start_date: String
    let end_date: String
    let max_volunteers: Int?
    let status: String
    let created_by: String
    let organization_id: String?
    let organization_name: String?
    let organization_logo_url: String?
    let organizer_name: String?
    let cover_photo_url: String?
    var photos: [EventPhoto]? = nil
    var fee: Double? = nil
    var is_free: Bool? = nil
    var latitude: Double? = nil
    var longitude: Double? = nil
    var waypoints: [Waypoint]? = nil

    static func == (lhs: Event, rhs: Event) -> Bool { lhs.id == rhs.id }
    func hash(into hasher: inout Hasher) { hasher.combine(id) }
}

struct EventListResponse: Codable {
    let items: [Event]
    let total: Int
}

File: backend/app/schemas/event.py
# filepath: backend/app/schemas/event.py
from datetime import datetime
from decimal import Decimal
from typing import Any, List, Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.models.event import EventCategory, EventDifficulty, EventStatus


class EventCreate(BaseModel):
    title: str
    description: Optional[str] = None
    category: EventCategory
    difficulty: EventDifficulty
    location_name: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    start_date: datetime
    end_date: datetime
    max_volunteers: Optional[int] = None
    requirements: Optional[dict] = None
    waypoints: Optional[List[dict]] = None
    route_geojson: Optional[Any] = None
    fee: Decimal = Field(default=Decimal("0.00"), ge=0, description="Katılım ücreti (TL). 0 ise ücretsiz.")


class EventUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    category: Optional[EventCategory] = None
    difficulty: Optional[EventDifficulty] = None
    status: Optional[EventStatus] = None
    location_name: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    max_volunteers: Optional[int] = None
    requirements: Optional[dict] = None
    waypoints: Optional[List[dict]] = None
    route_geojson: Optional[Any] = None


class EventUpdateFee(BaseModel):
    """Yalnızca ücret güncellemesi için kullanılır."""
    fee: Decimal = Field(..., ge=0, description="Yeni katılım ücreti (TL). 0 ise ücretsiz.")


class EventResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    title: str
    description: Optional[str] = None
    category: EventCategory
    difficulty: EventDifficulty
    status: EventStatus
    location_name: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    start_date: datetime
    end_date: datetime
    max_volunteers: Optional[int] = None
    requirements: Optional[dict] = None
    created_by: UUID
    organization_id: Optional[UUID] = None
    created_at: datetime
    organization_name: Optional[str] = None
    organization_logo_url: Optional[str] = None
    organizer_name: Optional[str] = None
    cover_photo_url: Optional[str] = None
    waypoints: Optional[List[dict]] = None
    route_geojson: Optional[Any] = None
    fee: float = 0.0
    is_free: bool = True


class EventListResponse(BaseModel):
    items: List[EventResponse]
    total: int
